lib/mini_portile2/mini_portile_cmake.rb

```diff
@@ -52,6 +51,77 @@ def cmake_cmd
 
   private
 
+  def generator_defaults
+    if MiniPortile.mswin? && generator_available?('NMake')
+      ['-G', 'NMake Makefiles']
+    elsif MiniPortile.mingw? && generator_available?('MSYS')
+      ['-G', 'MSYS Makefiles']
+    else
+      []
+    end
+  end
+
+  def cmake_compile_flags
+    c_compiler, cxx_compiler = find_c_and_cxx_compilers(host)
+
+    # needed to ensure cross-compilation with CMake targets the right CPU and compilers
+    [
+      "-DCMAKE_SYSTEM_NAME=#{cmake_system_name}",
+      "-DCMAKE_SYSTEM_PROCESSOR=#{MiniPortile.target_cpu}",
```

Comment:
I noticed on Windows builds, Ruby seems to return `x64` instead of `x86`. It doesn't seem to cause problems yet, but we might want to remap this automatically, or allow the user to override it.

Reply:
Yeah, this is something Nokogiri works around by [mapping the platform string to a tool prefix](https://github.com/sparklemotion/nokogiri/blob/main/rakelib/extensions.rake#L70-L71), and has been on my to-do list to move into either rake-compiler-dock or mini_portile for years. If it's not blocking you, I'm OK not worrying about it right now.